server/sonar-server/src/test/java/org/sonar/server/metric/ws/CreateActionTest.java

```diff
@@ -0,0 +1,236 @@
+      .setEnabled(false);
+    dbClient.metricDao().insert(dbSession, metricInDb);
+    dbSession.commit();
+
+    WsTester.Result result = newRequest()
+      .setParam(PARAM_KEY, DEFAULT_KEY)
+      .setParam(PARAM_NAME, DEFAULT_NAME)
+      .setParam(PARAM_TYPE, DEFAULT_TYPE)
+      .setParam(PARAM_DESCRIPTION, DEFAULT_DESCRIPTION)
+      .setParam(PARAM_DOMAIN, DEFAULT_DOMAIN)
+      .execute();
+
+    result.assertJson(getClass(), "metric.json");
+    result.outputAsString().matches("\"id\"\\s*:\\s*\"" + metricInDb.getId() + "\"");
+    MetricDto metricAfterWs = dbClient.metricDao().selectNullableByKey(dbSession, DEFAULT_KEY);
+    assertThat(metricAfterWs.getId()).isEqualTo(metricInDb.getId());
+    assertThat(metricAfterWs.getDomain()).isEqualTo(DEFAULT_DOMAIN);
+    assertThat(metricAfterWs.getDescription()).isEqualTo(DEFAULT_DESCRIPTION);
+    assertThat(metricAfterWs.getValueType()).isEqualTo(DEFAULT_TYPE);
+    assertThat(metricAfterWs.getShortName()).isEqualTo(DEFAULT_NAME);
+  }
+
+  @Test
+  public void fail_when_existing_activated_metric_with_same_key() throws Exception {
+    expectedException.expect(ServerException.class);
+    dbClient.metricDao().insert(dbSession, MetricTesting.newMetricDto()
+      .setKey(DEFAULT_KEY)
+      .setUserManaged(true)
+      .setEnabled(true));
+    dbSession.commit();
+
+    newRequest()
+      .setParam(PARAM_KEY, DEFAULT_KEY)
+      .setParam(PARAM_NAME, "any-name")
+      .setParam(PARAM_TYPE, DEFAULT_TYPE).execute();
+  }
+
+  @Test
+  public void fail_when_existing_non_custom_metric_with_same_key() throws Exception {
+    expectedException.expect(ServerException.class);
+    dbClient.metricDao().insert(dbSession, MetricTesting.newMetricDto()
+      .setKey(DEFAULT_KEY)
+      .setUserManaged(false)
+      .setEnabled(false));
+    dbSession.commit();
+
+    newRequest()
+      .setParam(PARAM_KEY, DEFAULT_KEY)
+      .setParam(PARAM_NAME, "any-name")
+      .setParam(PARAM_TYPE, DEFAULT_TYPE).execute();
+  }
+
+  @Test
+  public void fail_when_missing_key() throws Exception {
+    expectedException.expect(IllegalArgumentException.class);
+
+    newRequest()
+      .setParam(PARAM_NAME, DEFAULT_NAME)
+      .setParam(PARAM_TYPE, DEFAULT_TYPE).execute();
+  }
+
+  @Test
+  public void fail_when_missing_name() throws Exception {
+    expectedException.expect(IllegalArgumentException.class);
+
+    newRequest()
+      .setParam(PARAM_KEY, DEFAULT_KEY)
+      .setParam(PARAM_TYPE, DEFAULT_TYPE).execute();
+  }
+
+  @Test
+  public void fail_when_missing_type() throws Exception {
+    expectedException.expect(IllegalArgumentException.class);
+
+    newRequest()
+      .setParam(PARAM_NAME, DEFAULT_NAME)
+      .setParam(PARAM_KEY, DEFAULT_KEY).execute();
+  }
+
+  @Test
+  public void fail_when_insufficient_privileges() throws Exception {
+    expectedException.expect(ForbiddenException.class);
+    userSessionRule.login("login").setGlobalPermissions(GlobalPermissions.SCAN_EXECUTION);
```

Comment:
IMHO you can skip the ``setGlobalPermissions`` call: when you call ``login(String)``, the session is reset and all permissions are removed.